src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs

```diff
@@ -128,6 +128,11 @@ public interface ITypeSymbol : INamespaceOrTypeSymbol
         /// </summary>
         bool IsReadOnly { get; }
 
+        /// <summary>
+        /// True if the type is a record.
+        /// </summary>
+        bool IsRecord { get; }
```

Comment:
I'm very uncomfortable exposing this API. We tried very hard in C# 9 to make records appear indistinguishable from classes in metadata, and while we didn't get quite there yet, we do intend to get all the way there. Currently, it's implemented by finding the `Clone` member from metadata: this is very likely not going to be resilient to future changes around `with`, and it could mean that what `IsRecord` returns will change in the future. The only way I'd be comfortable with this is if was solely based on whether this is coming from source, and was defined by a record syntax node.

Reply:
> I'm very uncomfortable exposing this API. We tried very hard in C# 9 to make records appear indistinguishable from classes in metadata, 

We definitely did not do that in Roslyn itself though.  With SymbolDisplay showing that something is a record.  However, that currently uses internal-only bits of the compielr to function, making it so that hte rest of the IDE cannot give the same experience as this one compiler API.  We need a mechanism that allows us to do teh same so we can treat records consistently across our featureset.

Reply:
> Currently, it's implemented by finding the Clone member from metadata: this is very likely not going to be resilient to future changes around with,

Pretty sure we established the opposite in LDM: the Clone method is here to stay forever no matter what changes we make in the future. 

https://github.com/dotnet/csharplang/blob/464e8d16e31e211904474bdf58979e361ad5b667/meetings/2020/LDM-2020-07-01.md#conclusion-1

We have already told other customers that they can use that method signature in metadata as a method of identifying record types. 

@jcouv who has been driving these types of customer interactions.

Reply:
> Pretty sure we established the opposite in LDM: the Clone method is here to stay forever no matter what changes we make in the future.

The point is not that said method might go away. The point is that said method might not be a distinguishing characteristic of records.

> We have already told other customers that they can use that method signature in metadata as a method of identifying record types.

Who? Where? I certainly have not. In fact, I have explicitly cautioned customers that this method is not necessarily a future-proof way of detecting records, given that we may choose to implement a generalized way for any class to declare that method.

Reply:
> Who? Where? I certainly have not. In fact, I have explicitly cautioned customers that this method is not necessarily a future-proof way of detecting records, given that we may choose to implement a generalized way for any class to declare that method.

I would literally be ok with us making this a public WellKnownMembername, and having us literally just look for that name to make any and all determinations here.

Reply:
> The point is that said method might not be a distinguishing characteristic of records.

This is how the compiler distinguishes records. The method can never go away because of compat concerns (confirmed at length in the LDM meeting). This is a distinguishing characteristic of records. 

> Who? Where? 

ASP.NET team as it impacts their serialization scenarios. They have concrete scenarios where understanding records are valuable. Part of the discussion is what lead to the LDM meeting confirming that the methods cannot go away because of compat concerns.

Reply:
So just to be clear, is it correct to assume that adding an `IsRecord` property to ITypeSymbol is appropriate in this PR?